### PR TITLE
[ENH] add yarikoptic's command line scatterplot tool

### PR DESCRIPTION
**Pull request by mvdoc**
So far just adapting @yarikoptic's code to pymvpa commandline parsing options + added option to save figures

### CODE REVIEW

**Comment by yarikoptic** [comment]
woohoo!  thanks! please merge current master and resolve conflicts

**Comment by yarikoptic** [comment]
and we would need at least some basic tests ;-)

**Comment by mvdoc** [comment]
Added some very very very basic tests...basically just smoke tests

**Comment by yarikoptic** [comment]
unittests rule:
```
FAIL: test_docstrings (mvpa2.tests.test_suite.SuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/PyMVPA/PyMVPA/mvpa2/tests/test_suite.py", line 96, in test_docstrings
    self.fail('\n'.join(sfailures))
AssertionError: Some items have missing docstrings:
 modules: scatter
----------------------------------------------------------------------

Ran 615 tests in 377.010s
FAILED (SKIP=21, failures=1)

The command "eval   $COVERAGE $NOSETESTS -v mvpa2 $DOCTESTS -s" exited with 1.
0.18s$ make testsuite
I: Running full testsuite
E: 'from mvpa2.misc.plot.scatter import' is missing from mvpa2.suite()
make: *** [testsuite] Error 1

```
;-)

File: mvpa2/misc/plot/scatter.py [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+
+    verbose(1, "total: %d union: %d%s intersection: %d x_only: %d y_only: %d%s"
+            % (len(nzsum),
+               np.sum(union),
+               mask is not None and ' masked: %d' % np.sum(mask) or '',
+               np.sum(intersection),
+               np.sum(xnoty), np.sum(ynotx),
+               msg))
+
+    if verbose.level > 1:
```
Comment (yarikoptic) [inline mvpa2/misc/plot/scatter.py]: please add a 'TODO:  make this one independent of verbose.level - intoduce some argument/handling' line

File: mvpa2/misc/plot/scatter.py [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+                         "got {0}".format(len(dataXd)))
+    dataXd = np.asanyarray(dataXd)      # TODO: allow to operate on list of arrays to not waste RAM/cycles
+    data = dataXd.reshape((2, -1))
+    if dataXd.ndim < 5:
+        ntimepoints = 1
+    elif dataXd.ndim == 5:
+        ntimepoints = dataXd.shape[-1]
+    else:
+        raise ValueError("Do not know how to handle data with %d dimensions" % (dataXd.ndim - 1))
+    if x_jitter or y_jitter:
```
Comment (yarikoptic) [inline mvpa2/misc/plot/scatter.py]: just smoke test this one please (the same invocation but with x_jitter=0.1 or so)

File: mvpa2/misc/plot/scatter.py [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+        cm = pl.matplotlib.cm.get_cmap(colors)
+    if cm and len(dataXd.shape) > dimcolor+1:
+        cm.set_under((1, 1, 1, 0.1))             # transparent what is not in range
+        # we need to get our indices back for those we are going to plot.  probably this is the least efficient way:
+        ndindices_all = np.array(list(np.ndindex(dataXd.shape[1:])))
+        ndindices_nz = ndindices_all[intersection]
+        # choose color based on dimcolor
+        dimcolor_len = float(dataXd.shape[1+dimcolor])
+        edgecolors = cm(((cm.N-1) * ndindices_nz[:, dimcolor] / dimcolor_len).astype(int))
+        if mask is not None:
```
Comment (yarikoptic) [inline mvpa2/misc/plot/scatter.py]: some mask'ed smoke test would be awesome... may be the one with jitter ;)

File: mvpa2/misc/plot/scatter.py [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+            same_range = True
+            # assign limits the numerical range
+            limits = (np.min( [minx, miny] ),  np.max( [maxx, maxy] ))
+        else:
+            raise ValueError("Do not know how to handle same_range=%r" % (limits,))
+    else:
+        same_range = True
+
+    # Let's now plot threshold lines if provided
+    if thresholds is not None:
```
Comment (yarikoptic) [inline mvpa2/misc/plot/scatter.py]: and provide some thresholds (any)

File: mvpa2/misc/plot/scatter.py [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+    else: #if f.endswith('.nii.gz') or f.endswith('.img') or f.endswith('.hdr'):
+        n = nb.load(f)
+        data = n.get_data()
+        # strip rudimentary 4th dimension
+        if len(data.shape) == 4 and data.shape[-1] == 1:
+            data = data[:, :, :, 0]
+    return data
+
+
+def plot_scatter_files(files,
```
Comment (yarikoptic) [inline mvpa2/misc/plot/scatter.py]: you could use e.g. those: mvpa2/data/haxby2001/sub001/anatomy/lowres00{1,2}.nii.gz ... probably openfmri thing could provide a concise path, but `pathjoin(pymvpa_dataroot, *('haxby2001/sub001/anatomy/lowres001.nii.gz'.split('/')))` should work ;)

**Comment by yarikoptic** [comment]
How do you feel @mvdoc --will add more tests (e.g. just beefing up verbose.level for the duration of the test would help, or just expose that 'verbose' as 'include_stats' parameter for proper resolution) or that would be it for now?

**Comment by mvdoc** [comment]
ok, will keep making it more perfect ;)

**Comment by mvdoc** [comment]
@yarikoptic success? ;)

**Comment by mvdoc** [comment]
@yarikoptic anything else?

**Comment by yarikoptic** [comment]
I think that should be good for now!  Let's merge!  THANK YOU!  Now try to use it extensively so we could solidify before release happen there are issues

**Pull request merged**